GuideFinder-Mac/GuideFinder_Mac_2020_SMM_PAMless.Rmd

```diff
@@ -0,0 +1,1365 @@
+# }
+# ScramblesDF_2<-data.frame(ScramblesDF, stringsAsFactors = FALSE)
+
+
+
+```
+
+
+##2.Set BLAST Path & File 
+Set the path for the system command based on the name of your file (e.g. 12228_FASTA.fna ) and the location of BLAST  
+__Example__: Substitute bolded information in MakeDatabase and RunBlast  
+
+MakeDatabase<-' __/Users/spotom/Documents/Temp/Guide-Finder-master/GuideFinder-Mac/makeblastdb__  -in __mutansexamplefastafile.txt__ -dbtype nucl -out GenomeDatabase'  
+
+RunBlast<-' __/Users/spotom/Documents/Temp/Guide-Finder-master/GuideFinder-Mac/blastn__ -task blastn-short -query BLASTguides.fasta -db GenomeDatabase -outfmt 6 -num_threads 4 -dust no -soft_masking false -max_target_seqs 5 -word_size 13 -out MatchGuides.blast'  
+
+ScrambleBlast <- '__/Users/spotom/Documents/Temp/Guide-Finder-master/GuideFinder-Mac/blastn__ -task blastn-short -query BLASTscrambles.fasta -db GenomeDatabase -outfmt 6 -num_threads 4 -dust no -soft_masking false -max_target_seqs 5 -word_size 11 -perc_identity .9 -out MatchScrambles.blast'
+
+Similarly, set the path for (optional) scramble guide creation. Scramble guides are 20nt sequences without a match to anywhere in your input genome and can act as a negative control for your experiments. For ScrambleBlast, only change the path to the location of BLASTn on your computer (should be same as "RunBlast" path). (The input and output file names, i.e. BLASTscrambles.fasta and MatchScrambles.blast SHOULD NOT be changed)
+
+```{r}
+MakeDatabase<-'makeblastdb -in E_coli_DH5alpha/E_coli_CP017100.txt -dbtype nucl -out GenomeDatabase'
+
+RunBlast<-'blastn -task blastn-short -query BLASTguides.fasta -db GenomeDatabase -outfmt 6 -num_threads 4 -dust no -soft_masking false -max_target_seqs 5 -word_size 13 -out MatchGuides.blast'
+
+ScrambleBlast<-'blastn -task blastn-short -query BLASTscrambles.fasta -db GenomeDatabase -outfmt 6 -num_threads 4 -dust no -soft_masking false -max_target_seqs 5 -word_size 11 -perc_identity .9 -out MatchScrambles.blast'
+
+```
+
+
+##3.Input FASTA File
+Add the name of the fasta file that contains the whole nucleotide sequence for your genome (not the file
+with the individual coding sequences); if this was a draft genome, be sure that you used the preprocessing
+script to concatenate the several contigs into one 
+__Example__: Substitute the bolded information for the name of your FASTA file (kept in working directory)  
+InputGenome<- readDNAStringSet( __"mutansexamplefastafile.txt"__)
+```{r, tidy=TRUE}
+InputGenome<- readDNAStringSet("E_coli_DH5alpha/E_coli_CP017100.txt")
+
+```
+
+##4. Define Putative Promoter Region 
+Set the desired putative promoter region, for example, setting "PromoterRegion" to 50 if you want the Guide Finder to search 50 bp upstream of the transcription start site for each gene to identify guides. If you do not want Guide Finder to make guides from the promoter region, set to 0. Set between 0 - 100. Do not set above 100 bp. 
```

Comment:
Could be worth adding a quick comment whether going PAMless effects the likelihood we would use the promoter region - for example, with PAMless we get so many guides, user may want to keep promotor region excluded (set to 0) unless they are looking at a specific gene that did not get guides, or are interested in the promoter specifically? Any "rules of thumb" we go by can be helpful to include, even if future users disagree, it provides a starting place for them to know why we chose what we did (and also can always be nice discussion points in your paper at the end of the summer :) )